src/fancyline/tty/vt100.cr

```diff
@@ -10,11 +24,18 @@ class Fancyline
         super()
       end
 
-      # FIXME: Properly get the terminal dimensions, don't rely on environment
-      #        variables that are outdated at the time of reading them.
+      def winsize
+        LibC.ioctl(0, LibC::TIOCGWINSZ, out winsize)
```

Comment:
Missing an error-check. Not sure yet if this should raise an exception, or just return `nil` so it can be checked in the caller without the exception overhead.